File: app/src/main/java/danb/speedrunbrowser/api/objects/LeaderboardRunEntry.java
package danb.speedrunbrowser.api.objects;

public class LeaderboardRunEntry {
    public int place;
    public Run run;

    public String getPlaceName() {

        if((place / 10) % 10 == 1)
            return place + "th";

        switch(place % 10) {
            case 1:
                return place + "st";
            case 2:
                return place + "nd";
            case 3:
                return place + "rd";
            default:
                return place + "th";
        }
    }
}

File: app/src/main/java/danb/speedrunbrowser/api/objects/LeaderboardRunEntry.kt
package danb.speedrunbrowser.api.objects

import danb.speedrunbrowser.utils.Util
import java.io.Serializable

data class LeaderboardRunEntry(
        val run: Run,
        val place: Int? = null,
        val obsolete: Boolean = false
) : Serializable {
    val placeName: String
    get() = Util.formatRank(place)
}
